### ISSUE TITLE
[policy][api-key] error[io.gravitee.gateway.policy.PolicyChainException: io.gravitee.gateway.policy.PolicyException: java.lang.reflect.InvocationTargetException

### ISSUE THREAD

**Issue by kevinimbrechts** [gravitee-io/issues#2095]
Hi,

I have installed latest docker images (1.24.0) and I have re-created some APIs by importing files, but for the others I have had to re-create everything by hand because of errors.

Some of APIs returns 500 error. All my APIs have an API Key (I have tried keyless but I have the same error with "key-less").

### Expected Behavior
All APIs must work.

### Current Behavior
A blank response with 500 internal server error. (like https://github.com/gravitee-io/issues/issues/1888).

### Logs
```
An error occurs in policy[api-key] error[io.gravitee.gateway.policy.PolicyChainException: io.gravitee.gateway.policy.PolicyException: java.lang.reflect.InvocationTargetException
	at io.gravitee.gateway.policy.impl.RequestPolicyChain.execute(RequestPolicyChain.java:47)
	at io.gravitee.gateway.policy.impl.PolicyChain.doNext(PolicyChain.java:70)
	at io.gravitee.gateway.policy.impl.StreamablePolicyChain.doNext(StreamablePolicyChain.java:54)
	at io.gravitee.gateway.policy.impl.PolicyChain.handle(PolicyChain.java:102)
	at io.gravitee.gateway.policy.impl.PolicyChain.handle(PolicyChain.java:42)
	at io.gravitee.gateway.core.processor.chain.AbstractStreamableProcessorChain.handle(AbstractStreamableProcessorChain.java:59)
	at io.gravitee.gateway.handlers.api.ApiReactorHandler.handleClientRequest(ApiReactorHandler.java:112)
	at io.gravitee.gateway.handlers.api.ApiReactorHandler.doHandle(ApiReactorHandler.java:98)
	at io.gravitee.gateway.reactor.handler.AbstractReactorHandler.handle(AbstractReactorHandler.java:92)
	at io.gravitee.gateway.reactor.handler.AbstractReactorHandler.handle(AbstractReactorHandler.java:41)
	at io.gravitee.gateway.reactor.impl.DefaultReactor.lambda$route$2(DefaultReactor.java:101)
	at io.gravitee.gateway.core.processor.chain.AbstractProcessorChain.handle(AbstractProcessorChain.java:46)
	at io.gravitee.gateway.core.processor.chain.AbstractProcessorChain.lambda$handle$0(AbstractProcessorChain.java:41)
	at io.gravitee.gateway.reactor.processor.transaction.TransactionProcessor.handle(TransactionProcessor.java:55)
	at io.gravitee.gateway.reactor.processor.transaction.TransactionProcessor.handle(TransactionProcessor.java:29)
	at io.gravitee.gateway.core.processor.chain.AbstractProcessorChain.handle(AbstractProcessorChain.java:44)
	at io.gravitee.gateway.reactor.impl.DefaultReactor.route(DefaultReactor.java:108)
	at io.gravitee.gateway.standalone.vertx.VertxReactorHandler.handle(VertxReactorHandler.java:42)
	at io.gravitee.gateway.standalone.vertx.VertxReactorHandler.handle(VertxReactorHandler.java:29)
	at io.vertx.core.http.impl.HttpServerRequestImpl.handleBegin(HttpServerRequestImpl.java:150)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:129)
	at io.vertx.core.net.impl.ConnectionBase.handleRead(ConnectionBase.java:390)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:174)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.vertx.core.http.impl.Http1xUpgradeToH2CHandler.channelRead(Http1xUpgradeToH2CHandler.java:98)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.gravitee.gateway.policy.PolicyException: java.lang.reflect.InvocationTargetException
	at io.gravitee.gateway.policy.impl.PolicyImpl.invoke(PolicyImpl.java:97)
	at io.gravitee.gateway.policy.impl.PolicyImpl.onRequest(PolicyImpl.java:50)
	at io.gravitee.gateway.policy.impl.RequestPolicyChain.execute(RequestPolicyChain.java:45)
	... 54 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.gravitee.gateway.policy.impl.PolicyImpl.invoke(PolicyImpl.java:95)
	... 56 more
Caused by: java.lang.NullPointerException
	at io.gravitee.gateway.handlers.api.policy.api.ApiPolicyChainResolver.calculate(ApiPolicyChainResolver.java:48)
	at io.gravitee.gateway.policy.AbstractPolicyChainResolver.resolve(AbstractPolicyChainResolver.java:46)
	at io.gravitee.gateway.policy.AbstractPolicyChainResolver.provide(AbstractPolicyChainResolver.java:66)
	at io.gravitee.gateway.policy.AbstractPolicyChainResolver.provide(AbstractPolicyChainResolver.java:36)
	at io.gravitee.gateway.core.processor.provider.StreamableProcessorProviderChain.next(StreamableProcessorProviderChain.java:38)
	at io.gravitee.gateway.core.processor.provider.StreamableProcessorProviderChain.next(StreamableProcessorProviderChain.java:28)
	at io.gravitee.gateway.core.processor.chain.AbstractStreamableProcessorChain.handle(AbstractStreamableProcessorChain.java:39)
	at io.gravitee.gateway.core.processor.chain.AbstractStreamableProcessorChain.lambda$handle$1(AbstractStreamableProcessorChain.java:55)
	at io.gravitee.gateway.policy.impl.PolicyChain.doNext(PolicyChain.java:86)
	at io.gravitee.gateway.policy.impl.PolicyChain.handle(PolicyChain.java:102)
	at io.gravitee.gateway.policy.impl.PolicyChain.handle(PolicyChain.java:42)
	at io.gravitee.gateway.core.processor.chain.AbstractStreamableProcessorChain.handle(AbstractStreamableProcessorChain.java:59)
	at io.gravitee.gateway.core.processor.chain.AbstractStreamableProcessorChain.lambda$handle$1(AbstractStreamableProcessorChain.java:55)
	at io.gravitee.gateway.policy.impl.PolicyChain.doNext(PolicyChain.java:86)
	at io.gravitee.gateway.policy.impl.StreamablePolicyChain.doNext(StreamablePolicyChain.java:54)
	at io.gravitee.policy.apikey.ApiKeyPolicy.onRequest(ApiKeyPolicy.java:94)
	... 61 more
]
```

### Steps to Reproduce (for bugs)
1. Create a new API
2. Create a new plan with API Key and publish
3. Subscribe with an application
4. Try to GET the API with the API Key.

### Your Environment
* Version used: 1.24.0

**Comment by kevinimbrechts** [comment]
I think the problem comes with a manual creation, when I import an API from a link. I don't have tried this for now, but I think I have this problem when I check "Create policies on paths".

**Comment by brasseld** [comment]
Does it mean that you're creating an API by importing a file ?
A swagger ? Which one ?
Can you provide the file you're importing ?

**Comment by brasseld** [comment]
Do you think you can reproduce on our demo environment ?

https://demo.gravitee.io/

**Comment by kevinimbrechts** [comment]
Thanks @brasseld.

I export all my APIs from 1.21 version in "Details" screen.
They are JSON files. Some APIs could be imported, others not.

I try to reproduce on your demo env (with a JSON file not working on our Gravitee).
Your Gravitee displays "API created" but it's wrong. My API is not created and, in Chrome's console, I have these errors : 
```
Transition Rejection($id: 2 type: 6, message: The transition errored, detail: Error: Transition Rejection($id: 1 type: 4, message: This transition is invalid, detail: The following parameter values are not valid for state 'management.apis.detail.portal.general': [apiId:"undefined"]))

Error: Transition Rejection($id: 1 type: 4, message: This transition is invalid, detail: The following parameter values are not valid for state 'management.apis.detail.portal.general': [apiId:"undefined"])
```

Here is the JSON file on attachment (I have change many information).
[cllientstest.zip](https://github.com/gravitee-io/issues/files/3021758/cllientstest.zip)

**Comment by brasseld** [comment]
Ok, so the problem is coming from the fact that you have a group named "default"  and an endpoint named "default".

Please have a look to: https://docs.gravitee.io/apim_installguide_migration.html#endpoints_name_restrictions

**Comment by brasseld** [comment]
The problem I'm seeing here is that the title of the issue is not relevant according to your final comment.

Is the problem relative to the exception with api-key policy or the fact that importing an API definition does not work (because of duplicate names)

**Comment by kevinimbrechts** [comment]
Thank you @brasseld.
Oh ok ! But it's really strange because the export comes from 1.21, and your link is about 1.20.13. In fact, the APIs where the import didn't working was effectively created in 1.20 version, but, when I upgraded to 1.21, I don't have any problem.

I think the problems are the same. This stack appears when I try to import an "old" API. Is that clearer?

**Comment by brasseld** [comment]
1.21.x does not contain the fix from 1.2O.13 because 1.21 is no more supported, so the fix has not been pushed to 1.21.x

That's the reason why you were able to export from 1.21 without any restrictions...

**Comment by kevinimbrechts** [comment]
Ok, I understand better. In this case, problem solved. Thank you so much !
